torch/lib/THD/base/init_methods/InitMethodEnv.cpp

```diff
@@ -37,40 +37,51 @@ std::tuple<std::string, port_type> load_worker_env() {
   return std::make_tuple(must_getenv(MASTER_ADDR_ENV), port);
 }
 
-rank_type load_rank_env() {
-  return convertToRank(std::stol(must_getenv(RANK_ENV)));
+rank_type getRank(int rank) {
+  const char *env_rank_str = std::getenv(RANK_ENV);
+  int env_rank = rank;
+  if (env_rank_str != nullptr)
+    env_rank = std::stol(env_rank_str);
+  if (rank != -1 && env_rank != rank)
+    throw std::runtime_error("rank specified both as an environmental variable "
+      "and to the initializer");
+
+  return convertToRank(env_rank);
+}
+
+rank_type getWorldSize(int world_size) {
+  const char *env_world_size_str = std::getenv(WORLD_SIZE_ENV);
+  int env_world_size = world_size;
+  if (env_world_size_str != nullptr)
+    env_world_size = std::stol(env_world_size_str);
+  if (world_size != -1 && env_world_size != world_size)
+    throw std::runtime_error("world_size specified both as an environmental variable "
+      "and to the initializer");
+
+  return convertToRank(env_world_size);
```

Comment:
Maybe make these templates? Code looks exactly the same (modulo error msgs and getenv arg)